### PR TITLE
Support list a specific status of job

### PR DESCRIPTION
**Pull request by maobaolong**
### What changes are proposed in this pull request?

We want to list RUNNING,  FAILED,  COMPLETED, status job from restful api.

### Why are the changes needed?

We can list RUNNING,  FAILED,  COMPLETED, status job through restful api. 

### Does this PR introduce any user facing changes?

No

### CODE REVIEW

File: conf/metrics.properties.template [hunk 1]
```diff
@@ -1,99 +0,0 @@
-#
```
Comment (yuzhu) [inline conf/metrics.properties.template:1]: why is this file removed?

Comment (maobaolong) [reply conf/metrics.properties.template]: Revert it

**Comment by maobaolong** [comment]
@yuzhu Thanks for remind me, revert the deleted template file, PTAL

File: job/server/src/main/java/alluxio/master/job/JobMasterClientRestServiceHandler.java [hunk 2]
```diff
@@ -127,16 +128,23 @@ public JobInfo call() throws Exception {
...
   @Path(ServiceConstants.LIST)
-  public Response list() {
+  public Response list(@QueryParam("status") final String status) {
     return RestUtils.call(new RestUtils.RestCallable<List<Long>>() {
       @Override
       public List<Long> call() throws Exception {
-        return mJobMaster.list(ListAllPOptions.getDefaultInstance());
+        if (status != null) {
+          return mJobMaster.list(ListAllPOptions.newBuilder()
+              .addStatus(Status.valueOf(status))
```
Comment (yuzhu) [inline job/server/src/main/java/alluxio/master/job/JobMasterClientRestServiceHandler.java:143]: the api actually supports filtering jobs by multiple statuses, do we want to support that? 
users might want both completed and failed jobs in one call. It is up to you though. 

also the api supports filtering by name as well, you may find it useful to add that. we can get all the replicate jobs or evict jobs that are completed, for example.

Comment (maobaolong) [reply job/server/src/main/java/alluxio/master/job/JobMasterClientRestServiceHandler.java:143]: This is a great idea, will support it.

**Comment by maobaolong** [comment]
@yuzhu I modified the restful api to be more powerful just like you motioned, PTAL. Thank you for let me know more about `ListAllPOptions`

**Review by yuzhu — APPROVED** [review]
LGTM, thanks for the improvement

**Comment by yuzhu** [comment]
alluxio-bot, merge this please
alluxio-bot, merge this please
